src/pykeen/nn/node_piece.py

```diff
@@ -565,6 +566,127 @@ def resolve_aggregation(
     return aggregation
 
 
+class TokenizationRepresentationModule(RepresentationModule):
+    """A module holding the result of tokenization."""
+
+    #: the token ID of the padding token
+    padding_idx: int
+
+    #: the token representations
+    tokens: RepresentationModule
```

Comment:
what about renaming this and `total_num_tokens` to `vocabulary_size` ? Right now it's a bit misleading - we have `num_tokens` arg in the constructor which specifies how many tokens per node to use, but the total number of tokens is pretty much what a vocabulary is - and it is easier to draw parallels to NLP tokenizers

Reply:
-> c027bbe